src/chat/panes/left-menu/controller.js

```diff
@@ -1,44 +1,43 @@
 module.exports = class LeftMenuController {
-    constructor(CardService) {
+    constructor(ChatService, CardService) {
         this.cardService = CardService;
-        this.archiveScreen = CardService.archiveScreen;
-        this.uptodateScreen = CardService.uptodateScreen;
-        this.setSearchItem = CardService.searchItem;
-        this.setSearchScreen = CardService.searchScreen;
-        this.recycleBinScreen = CardService.recycleBinScreen;
-        this.profileScreen = CardService.profileScreen;
-        this.userScreen = CardService.userScreen;
+        if (this.screenType == undefined) {
+            this.screenType = 1;
+        }
+        ChatService.owner.then(owner => {
+            this.owner = owner
+        });
         CardService.bind('UptodateScreen', () => {
-            this.uptodateScreen = CardService.uptodateScreen;
+            this.screenType = this.owner.ScreenType.UPTODATE;
         })
         CardService.bind('ArchiveScreen', () => {
-            this.archiveScreen = CardService.archiveScreen;
+            this.screenType = this.owner.ScreenType.ARCHIEVE;
         })
         CardService.bind('SearchScreen', () => {
             this.setSearchItem = CardService.searchItem;
-            this.setSearchScreen = CardService.searchScreen;
+            this.screenType = this.owner.ScreenType.SEARCH;
         })
         CardService.bind('RecycleBin', () => {
-            this.recycleBinScreen = CardService.recycleBinScreen;
+            this.screenType = this.owner.ScreenType.RECYCLEBIN;
         })
         CardService.bind('User', () => {
-            this.userScreen = CardService.userScreen;
+            this.screenType = this.owner.ScreenType.USER;
         })
         CardService.bind('Profile', () => {
-            this.profileScreen = CardService.profileScreen;
+            this.screenType = this.owner.ScreenType.PROFILE;
         })
     }
 
     setItem(item) {
         if (item == "Uptodate")
-            this.cardService.setUptodateScreen();
+            this.cardService.setScreenType(this.owner.ScreenType.UPTODATE);

```

Comment:
How about just having one line here, `this.cardService.setScreenType(item)` and getting rid of everything else?